utils/diag/gen_ncdiag.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+import argparse
+import numpy as np
+from gradsio import gradsctl
+from netCDF4 import Dataset
+import os, sys
+
+DEFAULT_MISSING_VALUE = 1.e20
+
+def write_ncdiag(fn, ctl, v3d, v2d):
+    fn = os.path.abspath(fn)
+
+    if os.path.exists(fn):
+        raise RuntimeError("{} already exists there. Exit...".format(fn))
+        sys.exit(1)
+
+    ds = Dataset(fn, mode='w' , format="NETCDF4_CLASSIC")
+
+    # Note: actually U/V does not use xh, yh due to Arakawa-C stagged grid, but it should not
+    # matter as long as our validation does not use u,v related obs
+    dim_x_name = "xh"
+    dim_y_name = "yh"
+    dim_z_name = "zl"
+    dim_t_name = "Time"
+    ds.createDimension(dim_x_name, ctl.nx)
+    ds.createDimension(dim_y_name, ctl.ny)
+    ds.createDimension(dim_z_name, ctl.nz)
+    ds.createDimension(dim_t_name, None)
+
+    var_u_name = "u"
+    var_v_name = "v"
+    var_s_name = "salt"
+    var_t_name = "temp"
+    var_h_name = "h"
+    v3d_shape = (dim_t_name, dim_z_name,dim_y_name,dim_x_name)
+    var_u = ds.createVariable(var_u_name, "f4", v3d_shape)
+    var_v = ds.createVariable(var_v_name, "f4", v3d_shape)
+    var_s = ds.createVariable(var_s_name, "f4", v3d_shape)
+    var_t = ds.createVariable(var_t_name, "f4", v3d_shape)
+    var_h = ds.createVariable(var_h_name, "f4", v3d_shape)
+        
+    var_ssh_name = "ssh"
+    var_sst_name = "sst"
+    var_sss_name = "sss"
+    var_eta_name = "eta"
+    var_mld_name = "mld"
+    var_ssu_name = "ssu" # additional variables
+    var_ssv_name = "ssv"
+    v2d_shape = (dim_t_name, dim_y_name,dim_x_name)
+    var_ssh = ds.createVariable(var_ssh_name, "f4", v2d_shape)
+    var_sst = ds.createVariable(var_sst_name, "f4", v2d_shape)
+    var_sss = ds.createVariable(var_sss_name, "f4", v2d_shape)
+    var_eta = ds.createVariable(var_eta_name, "f4", v2d_shape)
+    var_mld = ds.createVariable(var_mld_name, "f4", v2d_shape)
+
+    var_ssu = ds.createVariable(var_ssu_name, "f4", v2d_shape)
+    var_ssv = ds.createVariable(var_ssv_name, "f4", v2d_shape)
+   
+    print("writing 3d vars")
+    var_u[0,:,:,:] = v3d['u'].astype("f4")
+    var_v[0,:,:,:] = v3d['v'].astype("f4")
+    var_s[0,:,:,:] = v3d['s'].astype("f4")
+    var_t[0,:,:,:] = v3d['t'].astype("f4")
+    var_h[0,:,:,:] = v3d['h'].astype("f4")
+
+    print("writing 2d vars")
+    var_ssh[0,:,:] = v2d['ssh'].astype("f4")
+    var_sst[0,:,:] = v2d['sst'].astype("f4")
+    var_sss[0,:,:] = v2d['sss'].astype("f4")
+    var_eta[0,:,:] = v2d['eta'].astype("f4")
+    var_mld[0,:,:] = v2d['mld'].astype("f4")
+    var_ssu[0,:,:] = v2d['ssu'].astype("f4")
+    var_ssv[0,:,:] = v2d['ssv'].astype("f4")
+
+    ds.close()
+
+    # add missing values, since they will be read by read_diag for obsop
+    ds = Dataset(fn, mode='r+' , format="NETCDF4_CLASSIC")
+    varlist = ["u","v","salt","temp","h","ssh","sst","sss","eta","mld","ssu","ssv"]
+    for var in varlist:
```

Comment:
@gmao-cda take a look at https://github.com/raphaeldussin/remap_WOA18/blob/main/Process_WOA2018.ipynb

on how to better work with xarray